component/src/main/java/com/celements/structEditor/fields/HiddenTagPageType.java

```diff
@@ -42,9 +42,20 @@ public void collectAttributes(AttributeBuilder attrBuilder, DocumentReference ce
     attrBuilder.addNonEmptyAttribute("type", "hidden");
     try {
       XWikiDocument cellDoc = modelAccess.getDocument(cellDocRef);
-      attrBuilder.addNonEmptyAttribute("name", modelAccess.getFieldValue(cellDoc, FIELD_NAME).or(
-          cellDocRef.getName()));
-      attrBuilder.addNonEmptyAttribute("value", getVelocityFieldValue(cellDoc, FIELD_VALUE).or(""));
+      String value = getVelocityFieldValue(cellDoc, FIELD_VALUE).or("");
+      String name = modelAccess.getFieldValue(cellDoc, FIELD_NAME).or(cellDocRef.getName());
+      Optional<String> cellValue = getStructDataEditorService().getCellValueAsString(cellDocRef,
+          modelContext.getDoc());
+      if (cellValue.isPresent()) {
+        value = cellValue.get();
+        Optional<String> attrName = getStructDataEditorService().getAttributeName(cellDoc,
+            modelContext.getDoc());
+        if (attrName.isPresent()) {
+          name = attrName.get();
+        }
+      }
```

Comment:
to distinguish if data is set in the `StructEditFieldClass`, please introduce in `StructuredDataEditorService`:
```
@Override
public boolean hasEditField(XWikiDocument cellDoc) {
  return getCellFieldName(cellDoc).isPresent();
}
```
if true use `getAttributeName(...).or("")` and `getCellValueAsString(...).or("")`
else use static hidden field `FIELD_NAME` and `FIELD_VALUE`